packages/next/src/build/webpack/plugins/define-env-plugin.ts

```diff
@@ -169,6 +169,7 @@ export function getDefineEnv({
         }),
     'process.turbopack': isTurbopack,
     'process.env.TURBOPACK': isTurbopack,
+    ...(isTurbopack ? config.experimental.turbo?.define : {}),
```

Comment:
I think this needs to be handled slightly differently to make sure it doesn't add keys that conflict with built-in features. Something like this to error:

```tsx
const userDefines = isTurbopack ? config.experimental.turbo?.define : {}
for (const key in userDefines) {
	if(defineEnv.hasOwnProperty(key)) {
		throw new Error('descriptive error message about the defined key trying to override a built-in value')
	}
        
   	defineEnv[key] = userDefines[key]
}
```